File: second_year/high_level_methods/Lesson2/src/time/Time.java
package time;

public final class Time extends TimeField {

    /**
     * Часы
     */
    private int hour;

    /**
     * Минуты
     */
    private int minute;

    /**
     * Секунды
     */
    private int second;

    public Time(int hour, int minute, int second) {
        validateHourFiled(hour);
        this.hour =  hour;

        validateMinuteFiled(minute);
        this.minute =  minute;

        validateSecondFiled(second);
        this.second =  second;
    }

    public Time(int hour, int minute) {
        validateHourFiled(hour);
        this.hour = hour;

        validateMinuteFiled(minute);
        this.minute =  minute;

        this.second = 0;
    }

    public Time(int hour) {
        validateHourFiled(hour);
        this.hour =  hour;

        this.minute = 0;
        this.second = 0;
    }

    public Time() {
        this.hour = 0;
        this.minute = 0;
        this.second = 0;
    }


    public int getHour() {
        return hour;
    }

    public int getMinute() {
        return minute;
    }

    public int getSecond() {
        return second;
    }


    public void setHour(int hour) {
        validateHourFiled(hour);
        this.hour =  hour;
    }

    public void setMinute(int minute) {
        validateMinuteFiled(minute);
        this.minute =  minute;
    }

    public void setSecond(int second) {
        validateSecondFiled(second);
        this.second =  second;
    }

    @Override
    public String toString() {
        return (this.hour < 10 ? "0" : "") + this.hour +
            (this.minute < 10 ? ":0" : ":") + this.minute +
            (this.second < 10 ? ":0" : ":") + this.second;
    }
}

File: second_year/high_level_methods/Lesson2/Time/Time.cs
namespace Time;

public sealed class Time
{
    private int _second;
    private int _minute;
    private int _hour;

    public Time(int hour = 0, int minute = 0, int second = 0)
    {
        Hour = hour;
        Minute = minute;
        Second = second;
    }

    public int Second
    {
        get => _second;
        set
        {
            if (value < 0 || value > 59)
            {
                throw new ArgumentOutOfRangeException("Недопустимое значение секунд: " + value +
                                                      " (допустимое 0 - 59)");
            }

            _second = value;
        }
    }

    public int Minute
    {
        get => _minute;
        set
        {
            if (value < 0 || value > 59)
            {
                throw new ArgumentOutOfRangeException("Недопустимое значение минут: " + value +
                                                      " (допустимое 0 - 59)");
            }

            _minute = value;
        }
    }

    public int Hour
    {
        get => _hour;
        set
        {
            if (value < 0 || value > 23)
            {
                throw new ArgumentOutOfRangeException("Недопустимое значение часа: " + value +
                                                      " (допустимое 0 - 23)");
            }

            _hour = value;
        }
    }

    public override string ToString()
    {
        return $"{_hour:D2}:{_minute:D2}:{_second:D2}";
    }
}